bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusGroupHandler.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cbus.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+import org.slf4j.Logger;
+
+import com.daveoxley.cbus.Application;
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.Group;
+import com.daveoxley.cbus.Network;
+
+/**
+ * The {@link CBusGroupHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+
+@NonNullByDefault
+public abstract class CBusGroupHandler extends BaseThingHandler {
+
+    protected final Logger logger;
+    protected @Nullable CBusNetworkHandler cBusNetworkHandler = null;
+    protected @Nullable Group group = null;
+    protected int applicationId = -1;
+    protected int groupId = -1;
+
+    public CBusGroupHandler(Thing thing, int applicationId, Logger logger) {
+        super(thing);
+        this.applicationId = applicationId;
+        this.logger = logger;
+    }
+
+    @Override
+    public abstract void handleCommand(ChannelUID channelUID, Command command);
+
+    @SuppressWarnings({ "null", "unused" })
+    @Override
+    public void initialize() {
+        /*
+         * Group Id has moved from config to property. Copy for backward compatibility
+         */
+        if (getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID) == null) {
```

Comment:
Although if you want a dumb workaround you can make a utility method then call it.
```
private static boolean isNull(@Nullable Object obj){
    return obj == null;
}
```
```suggestion
        if (isNull(getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID)) {
```

Reply:
As stupid as it is, I would prefer you use the workaround method instead of suppressing warnings. But I think we can both agree that it is stupid to need it in the first place.

Reply:
Updated to avoid this problem.